android/TodokTodok/app/src/main/java/com/team/todoktodok/presentation/view/profile/joined/vm/JoinedDiscussionsViewModelFactory.kt

```
@@ -0,0 +1,16 @@
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.team.domain.repository.MemberRepository
+
+class JoinedDiscussionsViewModelFactory(
+    private val memberRepository: MemberRepository,
+) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(JoinedDiscussionsViewModel::class.java)) {
+            return JoinedDiscussionsViewModel(memberRepository) as T
+        }
+        throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)